Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp

```diff
@@ -195,64 +195,56 @@ class PageLoaderClient final : public API::InjectedBundle::PageLoaderClient {
 
     void didStartProvisionalLoadForFrame(WebPage&, WebFrame& frame, RefPtr<API::Object>&) override
     {
-        auto* webKitFrame = webkitFrameGet(&frame);
-        if (!webKitFrame && !frame.isMainFrame())
```

Comment:
I think the whole problem comes from an error in this condition. I think it should be
if (!webKitFrame || !frame.isMainFrame())
and it should fix the problem. Can you check whether that fixes the issue for you?
There are other instances of this same condition that should be fixed as well. I can do that if you confirm that this works for you.